### PR TITLE
Exposure pipeline

### PR DESCRIPTION
**Pull request by t-downing**
Pipeline to calculate flood exposure using Worldpop and Floodscan. Basic methodology is to:

1. Calculate exposure raster (with `src.datasources.floodscan.calculate_recent_flood_exposure_rasters()`)
   1. Filter Floodscan raster to ≥ 0.05 (i.e. only keep pixels with more than 5% flooding, to reduce noise)
   2. Interpolate Floodscan raster to Worldpop grid
   3. Multiply Floodscan raster by Worldpop raster to get exposure raster
5. Take raster stats (currently just sum, with `src.datasources.floodscan.calculate_recent_flood_exposure_rasterstats()`)
   1. Iterate over admin2s and clip raster to calculate sum

Only things that need looking at are the actual functions used by the pipeline (i.e. what is outlined above). There are a couple of notebooks that may be of interest, whose functionality hasn't yet been integrated into either the pipeline or the [app](https://github.com/OCHA-DAP/ds-floodexposure-monitoring-app):

- `exposure_plotting`: the first two plots have already been integrated into the app, but the admin bounds ones haven't been yet. I think these could be pretty useful for picking out where specifically flooding is high.
- `floodscan_historical`: just calculating the 1998-2023 flood exposure using the `.nc` in the Google Drive (faster than stacking up all the historical COGs), which needs to be done whenever a new country is added (takes about two hours).

### CODE REVIEW

**Comment by hannahker** [comment]
@t-downing what kind of feedback do you think makes the most sense here? There's a convo to be had about what we want our "production" setup to be, but I think that's probably best had elsewhere.

**Comment by t-downing** [comment]
@hannahker yes good question, I should have specified. I think here it would be good just to agree on the core methods for:

1. calculating the exposure rasters
2. taking the raster stats of the exposure

Let me just point out where exactly I'm talking about.

File: src/datasources/floodscan.py [hunk 1]
```diff
@@ -19,7 +23,240 @@
...
+    if not das:
+        if verbose:
+            print("no new floodscan data to process")
+        return
+    ds_recent = xr.concat(das, dim="date")
+    # filter to only pixels with flood extent > 5% to reduce noise
+    ds_recent_filtered = ds_recent.where(ds_recent >= 0.05)
+    # interpolate to Worldpop grid and
+    # multiply by population to get exposure
+    exposure = ds_recent_filtered.interp_like(pop, method="nearest") * pop
```
Comment (t-downing) [inline src/datasources/floodscan.py:101]: Here is where we are actually calculating the exposure rasters. I am unsure whether it makes sense to only take pixels with flood extent ≥ 5%. I think we lose a fair amount of information this way, and I'm not sure we really benefit from the reducing the noise.

We also may want to think about whether to multiply by the relevant population raster for _that year_. Also, which population raster to use? There are several options on WorldPop (of which we're using the `2020_1km_Aggregated_UNadj`). There is also [GHSL](https://human-settlement.emergency.copernicus.eu/ghs_pop2023.php).

File: src/datasources/floodscan.py [hunk 2]
```diff
@@ -19,7 +23,240 @@
...
+            dff = (
+                da_clip.sum(dim=["x", "y"])
+                .to_dataframe(name="total_exposed")["total_exposed"]
+                .astype(int)
+                .reset_index()
+            )
+            dff["ADM2_PCODE"] = pcode
+            dfs.append(dff)
+
+        df_exp_adm_new = pd.concat(dfs, ignore_index=True)
```
Comment (t-downing) [inline src/datasources/floodscan.py:235]: And here is where we're calculating the raster stats (in this case, just `sum`). But I guess we should replace this with whatever standard method we are using in `ds-raster-stats`

Comment (hannahker) [reply src/datasources/floodscan.py:235]: Would we want to also upsample the raster as we're taking the raster stats here? I know we're going to the 1km worldpop grid, but if we're calculating to the admin 2 level we might want to go a bit more granular.

File: src/datasources/floodscan.py (hunk 1 above)
Comment (hannahker) [reply src/datasources/floodscan.py:101]: I don't have a lot of background in working with the Floodscan data, but I'd lean more towards removing that 5% threshold. I think we're already doing a lot of smoothing, so this seems a little bit over cautious to remove noise. We'd also want to be able to justify why we picked that 5% number, which seems slightly arbitrary.

Comment (t-downing) [reply src/datasources/floodscan.py:101]: As discussed- we will ditch the 5% and re-run the historical pipelines later #14

File: src/datasources/floodscan.py (hunk 2 above)
Comment (t-downing) [reply src/datasources/floodscan.py:235]: Yeah potentially worth considering in the future, but for now I think 1km is ok (all the other raster stats are 0.05deg, so around 5km).

File: .github/workflows/run_update_exposure.yml [hunk 3]
```diff
@@ -35,12 +33,13 @@ jobs:
...
-              owner: 'OCHA-DAP',
-              repo: 'ds-floodexposure-monitoring',
-              event_type: 'trigger_exposure_raster_stats'
-          })
+      env:
+        GITHUB_TOKEN: ${{ secrets.PAT }}
+      run: |
+        curl -L \
+          -X POST \
+          -H "Accept: application/vnd.github+json" \
```
Comment (hannahker) [inline .github/workflows/run_update_exposure.yml:41]: Darn, this looks like a tough one...

**Review by hannahker — APPROVED** [review]
LGTM!

File: .github/workflows/run_update_exposure.yml (hunk 3 above)
Comment (t-downing) [reply .github/workflows/run_update_exposure.yml:41]: Ah it's just taken from the [GH Action REST API documentation](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#create-a-repository-dispatch-event).

**Pull request merged**